### DISCUSSION TITLE
Adding support for schemas

### DISCUSSION THREAD

**Discussion by Akamatsu21** [permitio/cedar-agent#23] [Show and tell]
Hello, as per the discussion in [this thread](https://github.com/permitio/cedar-agent/discussions/7), myself and @tisauro have decided to make an attempt at implementing the schema support for cedar-agent. I have made a new branch for this which you can find [in this repo](https://github.com/Akamatsu21/cedar-agent). This is a rough plan I've been following with this:

- [x] Create the SchemaMemoryStore with get, update and delete operations
- [x] Implement creating schema from file
- [x] Add API endpoints for the three operations
- [x] Add validation to policy create/update endpoints
- [x] Add validation to entity create/update endpoints
- [x] Add validation of existing policies/entities to schema create/update endpoints
- [x] Write unit tests
- [x] Update documentation and examples

I've not created a pull request for now as the feature isn't finished yet (although I am happy to make one if it makes it easier to browse the changes?). I wanted to ask if one of the main contributors to this project would be able to spare some time and have a look through what I've done so far, to confirm that this is in line with your expectations and there aren't any glaring mistakes? I would consider myself intermediate at Rust, so I'd be very happy to learn if anything could have been done better.

**Comment by orweis** [comment]
Hi @Akamatsu21 - for sure @omer9564 will be able to review this once you have a PR, might take a little while to do so, but we can definitely make sure to cover it (CC: @obsd )

**Reply by Akamatsu21** [reply]
Hi @orweis and thank you for your reply. I have created a PR [here](https://github.com/permitio/cedar-agent/pull/24) with what I have for now. As the next step (validating entities) is looking a bit more complex than initially expected in terms of structuring it into the code, I figured I should have the branch in its current state reviewed first before I make more changes. Hope this is okay.

**Comment by omer9564** [comment]
@Akamatsu21 fixed it in the PR, it was merged and released as cedar-agent version `0.2.0` available both in crates.io & dockerhub

**Reply by Akamatsu21** [reply]
Thank you very much, feels good to see it all done now :)
